cpp/src/arrow/compute/kernels/scalar_cast_temporal.cc

```diff
@@ -421,14 +421,29 @@ struct CastFunctor<TimestampType, Date64Type> {
 // ----------------------------------------------------------------------
 // String to Timestamp
 
+template <typename ExpectZone>
```

Comment:
I don't think the templating is useful. Parsing the timestamp should be more costly than a mostly predictable branch.

Reply:
Removed the templating.

Filed ARROW-14581 for the Travis test failure.